frontend/src/app/trips/[id]/(tabs)/activities.tsx

```diff
@@ -1,47 +1,323 @@
+  //           location: {
+  //             citystate: activity.location.citystate,
+  //             latitude: activity.location.latitude,
+  //             longitude: activity.location.longitude,
+  //           },
+  //           notes: activity.notes,
+  //           netUpvotes: activity.netUpvotes,
+  //           isOnCalendar: activity.isOnCalendar,
+  //           category: activity.category,
+  //           rating: activity.rating,
+  //         }),
   //       },
-  //       // body: JSON.stringify(req),
-  //     });
-  //     if (!response.ok) {
-  //       throw new Error("Failed to fetch trip");
-  //     }
-  //     // Optionally, you can handle the response here
-  //     const data = await response.json();
-  //     setTrip(data);
-  //     console.log("Trip fetch:", data);
-  //   } catch (error: any) {
-  //     console.error("Error fetching trip:", error.toString());
+  //     );
+  //     return response.json();
+  //   });
+  //   try {
+  //     //const newActivities = await Promise.all(promises);
+  //     //setActivities(newActivities); // Update the activities state with new data including IDs
+  //   } catch (error) {
+  //     console.error("Error saving activities:", error);
   //   }
   // };
 
+  const handleSelectCategory = (category: CategoryKeys | "All") => {
+    setSelectedCategory(category);
+    if (category === "All") {
+      setFilteredActivities(activities);
+    } else {
+      const filtered = activities.filter((activity) =>
+        activity.category.some((type) => categories[category].includes(type)),
+      );
+      setFilteredActivities(filtered);
+    }
+  };
+
+  const handleSearch = (text: string) => {
+    setSearchTerm(text);
+    if (text) {
+      const searchedActivities = filteredActivities.filter((activity) =>
+        activity.name.toLowerCase().includes(text.toLowerCase()),
+      );
+      setFilteredActivities(searchedActivities);
+    } else {
+      setFilteredActivities(activities);
+    }
+  };
+
   return (
-    <View style={{ flex: 1, height: Dimensions.get("window").height }}>
-      {/* <Stack.Screen
-        options={{
-          title: '',
-          headerShown: true,
-          headerBackTitle: 'Back',
-          headerBackTitleStyle: { fontSize: 10 },
-          
-        }}
-      /> */}
+    <View
+      style={{
+        flex: 1,
+        height: Dimensions.get("window").height,
+        backgroundColor: "white",
+      }}
+    >
+      <View style={styles.searchContainer}>
+        <Feather name="search" size={20} color="black" />
+        <TextInput
+          placeholder="Search for activities..."
+          value={searchTerm}
+          onChangeText={handleSearch}
+          style={styles.searchInput}
+        />
+      </View>
+      <ScrollView
+        horizontal
+        contentContainerStyle={styles.categoryContainer}
+        showsHorizontalScrollIndicator={false}
+        alwaysBounceVertical={false}
+      >
+        <TouchableOpacity
```

Comment:
mấy cái touchableopacity dùng array map dc khom, có thể để mấy cái category trong array of object. Tham khảo này nèk https://stackoverflow.com/questions/61718955/how-to-access-a-single-element-from-touchable-opacity-inside-a-map-function